addon/components/sl-select.js

```diff
@@ -228,67 +321,48 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
                         Ember.warn( warning, !values.length );
                     }
 
-                    return callback(
-                        multiple ?
-                        filteredContent :
-                        Ember.get( filteredContent, 'firstObject' )
-                    );
-                },
-
-                minimumResultsForSearch: this.get( 'disableSearch' ) ? -1 : 0,
-
-                query: ( query ) => {
-                    const content = this.get( 'content' ) || [];
-                    const optionLabelPath = this.get( 'optionLabelPath' );
-                    const select2 = this;
-
-                    query.callback({
-                        results: content.reduce( ( results, item ) => {
-                            const typeOfItem = Ember.typeOf( item );
-                            const text = 'object' === typeOfItem ||
-                                'instance' === typeOfItem ?
-                                Ember.get( item, optionLabelPath ) :
-                                item;
-
-                            if (
-                                text &&
-                                select2.matcher( query.term, text.toString() )
-                            ) {
-                                results.push( item );
-                            }
-
-                            return results;
-                        }, [] )
-                    });
+                    if ( filteredContent.length > 0 ) {
+                        callback( multiple ? filteredContent : filteredContent[0] );
+                    } else {
+                        callback( [] );
+                    }
                 }
-            });
-
-            input.on( 'change', () => {
-                this.set( 'value', input.select2( 'val' ) );
-            });
-
-            const originalBodyOverflow = document.body.style.overflow || 'auto';
+            };
+        }
+    ),
 
-            input.on( 'select2-open', () => {
-                document.body.style.overflow = 'hidden';
-            });
+    /**
+     * Select2 query function
+     *
+     * @function
+     * @returns {Function}
+     */
+    select2Query: Ember.computed(
+        function() {
+            return ( query ) => {
+                const content = this.get( 'content' );
+                const optionLabelPath = this.get( 'optionLabelPath' );
+                const select2 = this.get( 'input' ).data( 'select2' ).opts;
```

Comment:
Like Eric mentioned in #434, it's due to a change in scope from #404 that was not fixed. I decided to just add the fix here instead of waiting to merge.